### PR TITLE
Windows compatibility

### PR DESCRIPTION
**Pull request by kjdoore**
Updated requirements and fixed default NUTS parameters to allow for Windows compatibility. Additionally, created simple tests for fitting a tutorial dataset as needed to resolve Issue #31.

### CODE REVIEW

File: ratingcurve/sklearn.py [hunk 1]
```diff
@@ -119,8 +125,18 @@ def _nuts_inference(self, **kwargs):
...
+
         nuts_kwargs = self.__default_nuts_kwargs.copy()
         nuts_kwargs.update(kwargs)
+
+        # Check to see if OS is Windows. If so, set cores=1 and print warning.
+        if sys.platform == 'win32' and nuts_kwargs['cores'] != 1:
+            nuts_kwargs.update({'cores': 1})
+            warnings.warn("PyMC NUTS sampler has known bug when fitting with "
+                          "more than one core on Windows. Overriding user "
+                          "input. Number of cores set to 1.", RuntimeWarning)
```
Comment (thodson-usgs) [inline ratingcurve/sklearn.py:138]: Let's show this warning no matter what, but not override.
Perhaps link back to the github issue, if one exists.
```suggestion
        # Check to see if OS is Windows. If so, set cores=1 and print warning.
        if sys.platform == 'win32':
            nuts_kwargs.update({'cores': 1})
            warnings.warn("PyMC NUTS sampler has known bug when using "
                          "multiple cores on Windows. On Windows, the "
                          "number of cores is set to 1 by default.", RuntimeWarning)
```

**Review by thodson-usgs — COMMENTED** [review]
Would you add windows to the testing matrix in .github/workflows/python-package.yml

I believe this should do it:
`   os: [ubuntu-latest, macos-latest, windows-latest]`

File: ratingcurve/sklearn.py (hunk 1 above)
Comment (kjdoore) [reply ratingcurve/sklearn.py:138]: So, you are thinking to let it crash if the user tries to set `cores != 1`?

Comment (thodson-usgs) [reply ratingcurve/sklearn.py:138]: I'm okay with that, so long as they see the warning. I'm okay with you overriding too, though it makes testing slightly harder.

The main thing is that users should see the warning whenever they fit nuts on Windows, so they know linux is better!

Comment (kjdoore) [reply ratingcurve/sklearn.py]: Sounds good!

**Pull request merged**